lib/postgrex/extensions/tsvector.ex

```diff
@@ -0,0 +1,96 @@
+defmodule Postgrex.Extensions.TSVector do
+  @moduledoc false
+
+  import Postgrex.BinaryUtils, warn: false
+  use Bitwise
+  use Postgrex.BinaryExtension, [send: "tsvectorsend"]
+
+  def encode(_) do
+    quote location: :keep do
+      values when is_list(values) ->
+        encoded_tsvectors = unquote(__MODULE__).encode_tsvector(values)
+        <<byte_size(encoded_tsvectors)::int32, encoded_tsvectors::binary>>
+      other ->
+        raise ArgumentError,
+        Postgrex.Utils.encode_msg(other, "a list of tsvectors")
+    end
+  end
+
+  def decode(_) do
+    quote do
+      <<len::int32, value::binary-size(len)>> ->
+        <<nb_lexemes::int32, words::binary>> = value
+        unquote(__MODULE__).decode_tsvector_values(:binary.bin_to_list(words))
```

Comment:
I think converting everything to a list is unnecessary. You can call `decode_tsvector_values(words)` and do the whole decoding using binaries:

```elixir
def decode_tsvector_values("") do
  []
end

def decode_tsvector_values(words) do
  [word, <<positions_count::16, rest::binary>>] = :binary.split(words, <<0>>)
  positions_bytes = positions_count * 2
  <<positions::binary-size(positions_bytes), rest::binary>> = rest
  positions = for <<weight::2, position::14 <- positions>>, do: {position, decode_weight(weight)}
  [%Postgrex.Lexeme{word: word, positions: positions} | decode_tsvector_values(rest)]
end

defp decode_weight(0), do: nil
defp decode_weight(1), do: :C
defp decode_weight(2), do: :B
defp decode_weight(3), do: :A
```

That should cover most of decoding but I haven't tested it.